compiler/rustc_trait_selection/src/traits/specialize/mod.rs

```diff
@@ -144,8 +144,22 @@ pub(super) fn specializes(tcx: TyCtxt<'_>, (impl1_def_id, impl2_def_id): (DefId,
     // taking advantage of upstream ones.
     let features = tcx.features();
     let specialization_enabled = features.specialization || features.min_specialization;
-    if !specialization_enabled && (impl1_def_id.is_local() || impl2_def_id.is_local()) {
-        return false;
+    if !specialization_enabled && impl1_def_id.is_local() {
```

Comment:
Could you leave a comment here explaining in prose what is being enforced? Just for clarity, for people scanning this in the future.